src/Microsoft.Extensions.ServiceDiscovery.Dns/DnsServiceEndPointResolverBase.cs

```diff
@@ -95,61 +94,29 @@ public async ValueTask<ResolutionStatus> ResolveAsync(ServiceEndPointCollectionS
             }
 
             endPoints.AddChangeToken(_lastChangeToken);
-            return _lastStatus;
+            return;
         }
     }
 
     private bool ShouldRefresh() => _lastEndPointCollection is null || _lastChangeToken is { HasChanged: true } || ElapsedSinceRefresh >= _nextRefreshPeriod;
 
     protected abstract Task ResolveAsyncCore();
 
-    private async Task ResolveAsyncInternal()
-    {
-        try
-        {
-            await ResolveAsyncCore().ConfigureAwait(false);
-        }
-        catch (Exception exception)
-        {
-            SetException(exception);
-            throw;
-        }
-
-    }
-
-    protected void SetException(Exception exception) => SetResult(endPoints: null, exception, validityPeriod: TimeSpan.Zero);
-
-    protected void SetResult(List<ServiceEndPoint> endPoints, TimeSpan validityPeriod) => SetResult(endPoints, exception: null, validityPeriod);
-
-    private void SetResult(List<ServiceEndPoint>? endPoints, Exception? exception, TimeSpan validityPeriod)
+    protected void SetResult(List<ServiceEndPoint> endPoints, TimeSpan validityPeriod)
     {
         lock (_lock)
         {
-            if (exception is not null)
-            {
-                _nextRefreshPeriod = GetRefreshPeriod();
-                if (_lastEndPointCollection is null)
-                {
-                    // Since end points have never been resolved, use a pending status to indicate that they might appear
-                    // soon and to retry for some period until they do.
-                    _lastStatus = ResolutionStatus.FromPending(exception);
-                }
-                else
-                {
-                    _lastStatus = ResolutionStatus.FromException(exception);
-                }
-            }
-            else if (endPoints is not { Count: > 0 })
+            if (endPoints is not { Count: > 0 })
```

Comment:
Is there a reason why you put the `not` version of this in the `if` instead of flipping it? Not a big deal but I think the other way would be slightly more readable.